internal/v1/handler_test.go

```diff
@@ -203,6 +203,77 @@ func TestGetComposeStatusNotFoundResponse(t *testing.T) {
 	}
 }
 
+func TestGetComposeStatusInternalServerErrorResponse(t *testing.T) {
+	ctx := context.Background()
+	composeId := uuid.New()
+	var composerStatus composer.ComposeStatus
+	apiSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if "Bearer" == r.Header.Get("Authorization") {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		require.Equal(t, "Bearer accesstoken", r.Header.Get("Authorization"))
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusInternalServerError)
+		err := json.NewEncoder(w).Encode(composerStatus)
+		require.NoError(t, err)
+	}))
+	defer apiSrv.Close()
+
+	dbase, err := dbc.NewDB()
+	require.NoError(t, err)
+	cr := ComposeRequest{
+		Distribution: "rhel-9",
+		Customizations: &Customizations{
+			// Since we are not calling handleCommonCompose but inserting directly to DB
+			// there is no point in using plaintext passwords
+			// If there is plaintext password in DB there is problem elsewhere (eg. CreateBlueprint)
+			Users: &[]User{
+				{
+					Name:     "user",
+					SshKey:   common.ToPtr("ssh-rsa AAAAB3NzaC2"),
+					Password: common.ToPtr("$6$password123"),
+				},
+			},
+		}}
+
+	crRaw, err := json.Marshal(cr)
+	require.NoError(t, err)
+	err = dbase.InsertCompose(ctx, composeId, "000000", "user000000@test.test", "000000", cr.ImageName, crRaw, (*string)(cr.ClientId), nil)
+	require.NoError(t, err)
+	srv, tokenSrv := startServer(t, &testServerClientsConf{ComposerURL: apiSrv.URL}, &ServerConfig{
+		DBase:            dbase,
+		DistributionsDir: "../../distributions",
+	})
+	defer func() {
+		err := srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	}()
+	defer tokenSrv.Close()
+	payloads := []struct {
+		composerStatus composer.ComposeStatus
+		imageStatus    ImageStatus
+	}{
+		{
+			composerStatus: composer.ComposeStatus{
+				ImageStatus: composer.ImageStatus{},
+			},
+		},
+	}
+	for idx, payload := range payloads {
+		fmt.Printf("TT payload %d\n", idx)
+		composerStatus = payload.composerStatus
+
+		respStatusCode, body := tutils.GetResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/composes/%s",
```

Comment:
maybe use the `srv.URL` here to, to be more flexible
```suggestion
		respStatusCode, body := tutils.GetResponseBody(t, fmt.Sprintf(srv.URL + "/api/image-builder/v1/composes/%s",
```

Reply:
the reason for `srv.URL` is that we might want to make the port dynamic to avoid crashes just due to port overlap… (or better: I'd like to make the port dynamic soonish :-D - like in [osbuild-composer tests](https://github.com/osbuild/osbuild-composer/blob/main/cmd/osbuild-worker-executor/main_test.go#L56) )

Reply:
perefct idea